src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -589,6 +589,18 @@ private void GenerateResourceLoader(IndentedStringBuilder writer)
 		{
 			TryAnnotateWithGeneratorSource(writer);
 			writer.AppendLineInvariant($"global::Windows.ApplicationModel.Resources.ResourceLoader.DefaultLanguage = \"{_defaultLanguage}\";");
+			writer.AppendLineInvariant("var primaryLanguageOverride = global::Windows.Globalization.ApplicationLanguages.PrimaryLanguageOverride;");
+			using (writer.BlockInvariant("if (primaryLanguageOverride.Length > 0)"))
+			{
+				writer.AppendLineInvariant("var culture = new global::System.Globalization.CultureInfo(primaryLanguageOverride);");
+				// Undocumented implementation detail of .NET. Non-existing cultures have ThreeLetterWindowsLanguageName = ZZZ.
+				// UWP doesn't set such cultures as current cultures.
+				using (writer.BlockInvariant("if (culture.ThreeLetterWindowsLanguageName != \"ZZZ\")"))
+				{
+					writer.AppendLineInvariant("global::System.Globalization.CultureInfo.CurrentCulture = culture;");
+					writer.AppendLineInvariant("global::System.Globalization.CultureInfo.CurrentUICulture = culture;");
+				}
+			}
```

Comment:
This "as a whole" doesn't match UWP. In UWP, the CurrentCulture and CurrentUICulture are set BEFORE the first line of `Main` (`global::Windows.UI.Xaml.Application.Start((p) => new App());)` executes. I don't really know how they do it.

Curious to know how they could be doing that if someone knows? (even if we don't need to match UWP - just curious :smile:)

Reply:
Yeah it's much better in Application.Start. Fixed.